rhizome/host/spec/cloud_hypervisor_spec.rb

```diff
@@ -13,23 +13,72 @@
     end
 
     it "can download a firmware" do
+      allow(Arch).to receive(:sym).and_return(:x64)
       expect(File).to receive(:exist?).with("/opt/fw/CLOUDHV-202402.fd").and_return(false)
       expect(FileUtils).to receive(:mkdir_p).with("/opt/fw")
       f = instance_double(File)
       expect(f).to receive(:path).and_return("/opt/fw/CLOUDHV-202402.fd.tmp")
       expect(fw).to receive(:safe_write_to_file).with("/opt/fw/CLOUDHV-202402.fd").and_yield(f)
-      expect(fw).to receive(:curl_file).with("https://github.com/ubicloud/build-edk2-firmware/releases/download/edk2-stable202402-arm64/CLOUDHV-arm64.fd", "/opt/fw/CLOUDHV-202402.fd.tmp").and_return("thesha")
+      expect(fw).to receive(:curl_file).with("https://github.com/ubicloud/build-edk2-firmware/releases/download/edk2-stable202402-x64/CLOUDHV-x64.fd", "/opt/fw/CLOUDHV-202402.fd.tmp").and_return("thesha")
       fw.download
     end
 
     it "fails if sha is incorrect" do
+      allow(Arch).to receive(:sym).and_return(:x64)
       expect(File).to receive(:exist?).with("/opt/fw/CLOUDHV-202402.fd").and_return(false)
       expect(FileUtils).to receive(:mkdir_p).with("/opt/fw")
       f = instance_double(File)
       expect(f).to receive(:path).and_return("/opt/fw/CLOUDHV-202402.fd.tmp")
       expect(fw).to receive(:safe_write_to_file).with("/opt/fw/CLOUDHV-202402.fd").and_yield(f)
-      expect(fw).to receive(:curl_file).with("https://github.com/ubicloud/build-edk2-firmware/releases/download/edk2-stable202402-arm64/CLOUDHV-arm64.fd", "/opt/fw/CLOUDHV-202402.fd.tmp").and_return("anothersha")
+      expect(fw).to receive(:curl_file).with("https://github.com/ubicloud/build-edk2-firmware/releases/download/edk2-stable202402-x64/CLOUDHV-x64.fd", "/opt/fw/CLOUDHV-202402.fd.tmp").and_return("anothersha")
       expect { fw.download }.to raise_error("Invalid SHA-256 digest")
     end
   end
+
+  describe "#cloud hypervisor" do
+    subject(:ch) { CloudHypervisor::VersionClass.new("35.1", "sha_ch", "sha_remote") }
+
+    it "can use an existing version of cloud hypervisor" do
+      expect(File).to receive(:exist?).with("/opt/cloud-hypervisor/v35.1/cloud-hypervisor").and_return(true)
+      expect(File).to receive(:exist?).with("/opt/cloud-hypervisor/v35.1/ch-remote").and_return(true)
+      expect(ch).not_to receive(:curl_file)
+      ch.download
+    end
+
+    it "can download cloud hypervisor" do
```

Comment:
Man, these tests are pretty brutal, I've gotta think about this one a bit.

Reply:
made some minor changes. did that make the tests a little more gentle?

Reply:
Not...terribly, hah. I think I would consider isolating a subroutine and writing one brutal test for it, and then having the tests theoretically about sha checking be shorter.

For example, although `safe_write_for_file` is not tested (but perhaps it should be), it is partially mocked out in a couple of places that, if not for that, would unroll into a thicket of partial mocks in code that can do something like this:

```
rhizome/host/spec/cloud_hypervisor_spec.rb:      expect(fw).to receive(:safe_write_to_file).with("/opt/fw/CLOUDHV-202402.fd").and_yield(f)
```

Reply:
tests less brutal now?